wapitiCore/attack/mod_buster.py

```diff
@@ -99,6 +100,12 @@ async def test_directory(self, path: str):
 
         with open(path_join(self.DATA_DIR, self.PATHS_FILE), encoding="utf-8", errors="ignore") as wordlist:
             while True:
+                if (datetime.utcnow() - self.start).total_seconds() > self.max_attack_time >= 1:
```

Comment:
It will definitively work although I think the recommended solution would be to use `time.monotonic` to prevent changes in computer time.

Also it should be possible to leverage the `_stop_event` but it would be more difficult and would need more testing (something like creating a task that will set the event after X seconds, https://stackoverflow.com/questions/45419723/python-timer-with-asyncio-coroutine/45430833#45430833 ).

Maybe let's create another issue for that behavior.

I'm ok with the current MR solution but it requires more code in each module.

Reply:
@bretfourbe peux-tu faire la modif du `time.monotonic` ? 

Tu peux ignorer la partie `_stop_event` pour le moment, ce serait à faire dans une issue séparée sauf si tu dis que c'est trivial.

Reply:
I changed to `time.monotonic` :) 
@devl00p i think only few modules require this change since their execution can take a lot of time for each request